api/environments/identities/views.py

```diff
@@ -161,14 +162,18 @@ class SDKIdentities(SDKAPIView):
             cache=settings.GET_IDENTITIES_ENDPOINT_CACHE_NAME,
         )
     )
-    def get(self, request):  # type: ignore[no-untyped-def]
-        identifier = request.query_params.get("identifier")
-        if not identifier:
+    def get(self, request: Request) -> Response:
+        query_serializer = SDKIdentitiesQuerySerializer(data=request.query_params)
+        if not query_serializer.is_valid():
+            [error_message] = query_serializer.errors["identifier"]
```

Comment:
What happens if the query serializer has errors, but they are not caused by the identifier field?

Reply:
Good question. This is based on `SDKIdentitiesQuerySerializer` only having two fields:
- `identifier` (required)
- `transient` (optional, default = false)

I'd prefer to respond the request with all serializer errors, though the above decision is to keep a minimal similarity with existing error response message for this endpoint.

Reply:
My problem is that we are assuming whenever the serializer fails, it must have been caused by the identifier field. But since there are other fields in the serializer, if it fails because of those, this code will crash.
e.g: the following test: 

```
def test_SDKIdentities__identifier_sanitization__accepts_valid_identifiers(
    api_client: APIClient,
    environment: Environment,
    given_identifier: str,
) -> None:
    # Given
    api_client.credentials(HTTP_X_ENVIRONMENT_KEY=environment.api_key)

    # When
    response = api_client.get(
        f"/api/v1/identities/?identifier={quote(given_identifier)}&transient=meh"
    )

    # Then
    assert response.status_code == status.HTTP_200_OK
    assert response.json()["identifier"] == unquote(given_identifier)


```